patches/minecraft/net/minecraft/block/BlockRailBase.java.patch

```diff
@@ -39,10 +39,27 @@
 +    }
 +
 +    /**
++     * Return the rail's direction.
++     * Can be used to make the cart think the rail is a different shape,
++     * for example when making diamond junctions or switches.
++     * The cart parameter will often be null unless it it called from EntityMinecart.
++     *
++     * @param world The world.
++     * @param pos Block's position in world
++     * @param state The BlockState
++     * @param cart The cart asking for the metadata, null if it is not called by EntityMinecart.
++     * @return The direction.
++     */
++    public EnumRailDirection getRailDirection(IBlockAccess world, BlockPos pos, IBlockState state, @javax.annotation.Nullable net.minecraft.entity.item.EntityMinecart cart)
++    {
```

Comment:
Annotations are not actually a hard dependency. If a member has an annotation and that annotation class doesn't exist at runtime, the annotation data is simply dropped for that annotation.